src/basilisp/lang/interfaces.py

```diff
@@ -374,6 +374,18 @@ def pop(self: Self) -> Self:
         raise NotImplementedError()
 
 
+class IReduce:
+    @abstractmethod
+    def reduce(self, f, init):
```

Comment:
Is `init` optional? In the actual implementation it is.

Reply:
Yes. I annotated it with `Optional`.